pycbc/types/config.py

```diff
@@ -463,22 +463,34 @@ def add_options_to_section(self, section, items, overwrite_options=False):
 
     def sanity_check_subsections(self):
         """
-        This function goes through the ConfigParset and checks that any options
+        This function goes through the ConfigParser and checks that any options
         given in the [SECTION_NAME] section are not also given in any
         [SECTION_NAME-SUBSECTION] sections.
 
         """
         # Loop over the sections in the ini file
         for section in self.sections():
-            # [pegasus_profile] specially is allowed to be overriden by
+            # [pegasus_profile] is specially allowed to be overriden by
             # sub-sections
             if section == "pegasus_profile":
                 continue
 
+            if section.endswith('-defaultvalues') and \
+                    not len(section.split('-')) == 2:
+                # Only allow defaultvalues for top-level sections
+                raise NotImplementedError(
+                    "-defaultvalues subsections are only allowed for "
+                    "top-level sections; given %s" % section
+                )
+
             # Loop over the sections again
             for section2 in self.sections():
                 # Check if any are subsections of section
                 if section2.startswith(section + "-"):
+                    if section2.endswith("defaultvalues"):
```

Comment:
How can I get here now? Isn't this case going to trapped at line 481 and raise a failure?

Reply:
You would get here e.g. in the case that `section` is `coinc` and `section2` is `coinc-defaultvalues`.

The logic above catches when you try to provide defaults to a subsection, e.g. `coinc-fullinj-defaultvalues` through the `len()` check